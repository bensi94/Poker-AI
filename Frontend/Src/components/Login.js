import React from 'react';
import { Form, Input, Button, Layout } from 'element-react';


class Login extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            form: {
                username: '',
                password: ''
            },
            rules: {
                username: [
                    { required: true, message: 'Please input the username', trigger: 'blur' },
                    { validator: (rule, value, callback) => {
                        if (value === '') {
                            callback(new Error('Please input the username'));
                        } else {
                            callback();
                        }
                    } }
                ],
                password: [
                    { required: true, message: 'Please input the password ', trigger: 'blur' },
                    { validator: (rule, value, callback) => {
                        if (value === '') {
                            callback(new Error('Please input the password'));
                        } else {
                            callback();
                        }
                    } }
                ]
            }
        };
    }

    handleSubmit(e) {
        e.preventDefault();

        this.refs.form.validate((valid) => {
            if (valid) {
                console.log('isValid');
            } else {
                console.log('invalid');
                return false;
            }
        });
    }

    handleReset(e) {
        e.preventDefault();

        this.refs.form.resetFields();
    }

    onChange(key, value) {
        this.setState({
            form: Object.assign({}, this.state.form, { [key]: value })
        });
    }

    render() {
        return (
            <Layout.Row type="flex" justify="center">
                <Form ref="form"
                    model={this.state.form}
                    rules={this.state.rules}
                    labelWidth="100"
                    className="signup-login-form"
                >
                    <h2 className="signup-login-title">Login</h2>
                    <Form.Item label="Username" prop="username">
                        <Input type="text"
                            value={this.state.form.username}
                            onChange={this.onChange.bind(this, 'username')}
                            autoComplete="off"
                        />
                    </Form.Item>
                    <Form.Item label="Pasword" prop="password">
                        <Input type="password"
                            value={this.state.form.password}
                            onChange={this.onChange.bind(this, 'password')}
                            autoComplete="off"
                        />
                    </Form.Item>
                    <Form.Item>
                        <Button type="primary" onClick={this.handleSubmit.bind(this)}>Login</Button>
                        <Button onClick={this.handleReset.bind(this)}>Reset</Button>
                    </Form.Item>
                </Form>
            </Layout.Row>
        );
    }
}

export default Login;
