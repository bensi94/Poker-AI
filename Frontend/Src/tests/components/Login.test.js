import React from 'react';
import { shallow } from 'enzyme';

import Login from '../../components/Login';

describe('Login Test suite', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Login />);
    });

    it('Should render Login(snapshot) correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should change username', () => {
        const username = 'bensi94';

        // Input field at 0 should be username field
        wrapper.find('Input').at(0).simulate('change', username);

        expect(wrapper.state('form').username).toBe(username);
    });

    it('Should change password', () => {
        const password = 'testpassword';

        // Input field at 1 should be password field
        wrapper.find('Input').at(1).simulate('change', password);

        expect(wrapper.state('form').password).toBe(password);
    });

    it('Should call submit on button clicked', () => {
        // Replaces the handleSubmit with a mock function
        // that should be clicked when the button is clicked
        const submitMock = jest.fn();
        wrapper.instance().handleSubmit = submitMock;
        wrapper.instance().forceUpdate();
        wrapper.find('Button').at(0).simulate('click');

        expect(submitMock).toHaveBeenCalledTimes(1);
    });

    it('Should have a username that is not empty', () => {
        const username = '';
        const callback = jest.fn();

        // Calling the validator with an empty username
        wrapper.instance().state.rules.username[1].validator('', username, callback);

        expect(callback).toHaveBeenCalledTimes(1);
        expect(callback).toHaveBeenCalledWith(new Error('Please input the username'));
    });

    it('Should have a password that is not empty', () => {
        const password = '';
        const callback = jest.fn();

        // Calling the validator with an empty username
        wrapper.instance().state.rules.password[1].validator('', password, callback);

        expect(callback).toHaveBeenCalledTimes(1);
        expect(callback).toHaveBeenCalledWith(new Error('Please input the password'));
    });
});
