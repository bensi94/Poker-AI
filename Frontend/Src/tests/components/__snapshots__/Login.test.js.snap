// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Test suite Should render Login(snapshot) correctly 1`] = `
<Row
  align="top"
  justify="center"
  tag="div"
  type="flex"
>
  <Form
    className="signup-login-form"
    labelPosition="right"
    labelSuffix=""
    labelWidth="100"
    model={
      Object {
        "password": "",
        "username": "",
      }
    }
    rules={
      Object {
        "password": Array [
          Object {
            "message": "Please input the password ",
            "required": true,
            "trigger": "blur",
          },
          Object {
            "validator": [Function],
          },
        ],
        "username": Array [
          Object {
            "message": "Please input the username",
            "required": true,
            "trigger": "blur",
          },
          Object {
            "validator": [Function],
          },
        ],
      }
    }
  >
    <h2
      className="signup-login-title"
    >
      Login
    </h2>
    <FormItem
      label="Username"
      prop="username"
    >
      <Input
        autoComplete="off"
        autosize={false}
        onChange={[Function]}
        rows={2}
        trim={false}
        type="text"
        value=""
      />
    </FormItem>
    <FormItem
      label="Pasword"
      prop="password"
    >
      <Input
        autoComplete="off"
        autosize={false}
        onChange={[Function]}
        rows={2}
        trim={false}
        type="password"
        value=""
      />
    </FormItem>
    <FormItem>
      <Button
        disabled={false}
        loading={false}
        nativeType="button"
        onClick={[Function]}
        plain={false}
        type="primary"
      >
        Login
      </Button>
      <Button
        disabled={false}
        loading={false}
        nativeType="button"
        onClick={[Function]}
        plain={false}
        type="default"
      >
        Reset
      </Button>
    </FormItem>
  </Form>
</Row>
`;
